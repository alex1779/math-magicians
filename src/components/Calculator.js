import PropTypes from 'prop-types';

function Calculator() {
  return (

    <div className="default">

      <input type="text" placeholder="0" className="input-result" />

      <div className="buttons">

        <button className="buttons-group1" type="button">AC</button>
        <button className="buttons-group1" type="button">+/-</button>
        <button className="buttons-group1" type="button">%</button>
        <button className="buttons-group2" type="button">รท</button>

        <button className="buttons-group1" type="button">7</button>
        <button className="buttons-group1" type="button">8</button>
        <button className="buttons-group1" type="button">9</button>
        <button className="buttons-group2" type="button">x</button>

        <button className="buttons-group1" type="button">4</button>
        <button className="buttons-group1" type="button">5</button>
        <button className="buttons-group1" type="button">6</button>
        <button className="buttons-group2" type="button">-</button>

        <button className="buttons-group1" type="button">1</button>
        <button className="buttons-group1" type="button">2</button>
        <button className="buttons-group1" type="button">3</button>
        <button className="buttons-group2" type="button">+</button>

        <button className="buttons-group3" type="button">0</button>
        <button className="buttons-group1" type="button">.</button>
        <ButtonEqual string="=" />
      </div>

    </div>

  );
}

function ButtonEqual(props) {
  const { string } = props;
  return (
    <button className="buttons-group2" type="button">
      {string}
    </button>
  );
}

ButtonEqual.defaultProps = {
  string: 0,
};

ButtonEqual.propTypes = {
  string: PropTypes.string,
};

export default Calculator;
