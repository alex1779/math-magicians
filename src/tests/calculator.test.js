import renderer from 'react-test-renderer';
import React from 'react';
import Calculator from '../components/Calculator';

test('Confirm that the page renders correctly', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
