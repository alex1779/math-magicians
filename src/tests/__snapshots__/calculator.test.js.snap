// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm that the page renders correctly 1`] = `
<div
  className="intro"
>
  <p
    className="intro-p"
  >
    Let's do some math!
  </p>
  <div
    className="default"
  >
    <input
      className="input-result"
      placeholder="0"
      type="text"
      value="0"
    />
    <div
      className="buttons"
    >
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="buttons-group2"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="buttons-group2"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="buttons-group2"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="buttons-group2"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="buttons-group3"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="buttons-group1"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="buttons-group2"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
